codegen/src/main/scala/overflowdb/codegen/CodeGen.scala

```diff
@@ -1559,9 +1562,14 @@ class CodeGen(schema: Schema) {
       s"""package $nodesPackage
          |
          |/** base type for all nodes that can be added to a graph, e.g. the diffgraph */
-         |trait NewNode extends AbstractNode with Product {
+         |abstract class NewNode extends AbstractNode with overflowdb.DetachedNodeData with Product {
```

Comment:
It does compile, which is pretty cool!

But we don't need multiple inheritance of fields here? We can place `refOrId` at a constant offset (same offset for all NewNode implementations) inside the class and permit access without megamorphic virtual dispatch. We don't need scalac to synthesize separate getters and setters for each NewNode implementation and can instead have one shared implementation in the base class.

Consider `javap -p -s -c` of the scalac output of
 ```
package foo
trait NNT { var refOrId: Object = null }
class AT extends NNT

abstract class NNC {var refOrId:Object = null}

class AC extends NNC
```
Regardless, if you still think there are tangible advantages to using a trait instead of an abstract base class, then sure why not.